test/e2e/common/runtime.go

```diff
@@ -331,49 +331,90 @@ while true; do sleep 1; done
 				}
 			}
 
-			It("should not be able to pull image from invalid registry [NodeConformance]", func() {
+			/*
+				Release : v1.15
+				Testname: Container Runtime, image registry, pull from invalid registry
+				Description: Create a Pod with a container which is set to pull image from invalid registry. Container MUST be in waiting state due to error during image pull and the Pod status MUST be in Pending state.
+			*/
+			framework.ConformanceIt("should not be able to pull image from invalid registry [NodeConformance]", func() {
 				image := "invalid.com/invalid/alpine:3.1"
 				imagePullTest(image, false, v1.PodPending, true, false)
 			})
 
-			It("should not be able to pull non-existing image from gcr.io [NodeConformance]", func() {
+			/*
+				Release : v1.15
+				Testname: Container Runtime, image registry, pull non-existing image
+				Description: Create a Pod with a container which is set to pull non-existing image from the image registry. Container MUST be in waiting state due to error during image pull and the Pod status MUST be in Pending state.
+			*/
+			framework.ConformanceIt("should not be able to pull non-existing image from gcr.io [NodeConformance]", func() {
```

Comment:
We can't promote this, b/c conformance needs to be able to run in air-gapped environments, gcr.io may not even be accessible so the error would be different.

Reply:
Suggestion: use the image utils package to grab the registry where we were getting our e2e test images (usually gcr, but can be overridden in airgap situations or for other reason). Then you should know the registry is good and the user has a way to tweak it if they needed. You'll just build the image name manually using that registry name.

Reply:
Would this e2e promotable to Conformance if we update the codebase in another PR to use image util package to grab registry?
From @timothysc, this e2e is not promotable because -
1. gcr.io may not be accessible for a conformant platform.
2. Errors would be different and might fail with unexpected pull error. [[code-link](https://github.com/kubernetes/kubernetes/blob/a73730147c8b45ad37a56ba210a9c721ed7a6844/test/e2e/common/runtime.go#L298-L300)] (**If this is the case, should we remove checking against specific 'error reasons', instead just fail incase of any error occur**?)

In case, it is non Promotable, we can drop its promotion with appropriate comment.

Reply:
We could promote if we refactor to support an override for registry and modified the test to denote.